source/val/validate_type.cpp

```diff
@@ -246,19 +353,12 @@ spv_result_t ValidateTypeStruct(ValidationState_t& _, const Instruction* inst) {
              << ".";
     }
     if (_.IsForwardPointer(member_type_id)) {
-      // If we're dealing with a forward pointer:
-      // Find out the type that the pointer is pointing to (must be struct)
-      // word 3 is the <id> of the type being pointed to.
-      auto type_pointing_to = _.FindDef(member_type->words()[3]);
-      if (type_pointing_to && type_pointing_to->opcode() != SpvOpTypeStruct) {
-        // Forward declared operands of a struct may only point to a struct.
+      const auto inst_index = inst - &_.ordered_instructions()[0];
+      const auto member_index = inst - &_.ordered_instructions()[0];
```

Comment:
Is one of these indexes wrong? Otherwise, won't these always be the same value?

Reply:
Yes, thanks

Reply:
I removed this check. It gets covered by checking forward references.